### ISSUE TITLE
Heatmaps and tags in the layers

### ISSUE THREAD

**Issue by yezhengli-Mr9** [xingyizhou/ExtremeNet#21]
1. HEATMAPS: For figure 4 of [arXiv](https://arxiv.org/abs/1901.08043), I think you are using the heatmaps after softmax?
I mean not ```self.l_heats``` in ```_train(*)``` or ```_test(*)``` of ```exkp.py``` but heat maps after  ```_sigmoid(``` in ```CTLoss(*)``` of ```exkp.py```(when training) or ```_exct_decode(*)``` of ```kp_utils.py```(when testing). Is that correct?
For heatmaps before softmax, I am unable to obtain an explainable visualization; but my heatmaps after softmax seems urgly. (I need to double check my visualization).
2. TAGS in the layer: Compared to CornerNet, it seems counterparts for ```self.tl_tags``` ```self.dr_tags``` do not exist. Is this mentioned some where in the [arXiv](https://arxiv.org/abs/1901.08043)? I am not proficient enough to figure out the reason of the differences.

**Comment by xingyizhou** [comment]
Hi,
  Thanks for your detailed reading. 1: all the heatmaps used in the paper and code are after softmax. You can turn on the debug mode `python test.py ExtremeNet --debug` to visualize them. 2: tags are the associative embedding features in the CornerNet. We don't need them. We use different grouping technic (center grouping). The comparison between center grouping and AE grouping is shown in Table. 1 (`w/o center`).

**Comment by yezhengli-Mr9** [comment]
> Hi,
> Thanks for your detailed reading. 1: all the heatmaps used in the paper and code are after softmax. You can turn on the debug mode `python test.py ExtremeNet --debug` to visualize them. 2: tags are the associative embedding features in the CornerNet. We don't need them. We use different grouping technic (center grouping). The comparison between center grouping and AE grouping is shown in Table. 1 (`w/o center`).

Thanks a lot. Yep, I definitely tried --debug mode -- it seems include all the visualization technique. Yep, ```associative embedding``` is mentioned in your paper.
